### PR TITLE
assessment

### PR DESCRIPTION
**Pull request by Ahmedkhafaji**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ x] JavaScript Coding Challenges
- [x ] Ruby Coding Challenges
- [ x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,133 @@
...
+    return 1
+ }else if(letter === "o"|| letter === "O") {
+    return 0 
+}else
+  return letter
+ })
+.join("")
+}
+
+//Test Suites: 1 passed, 1 total
```
Comment (elysemontano) [inline code-challenges.test.js:63]: This looks wonderful Ahmed!!  Great work!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,133 @@
...
+
+
+const fullHouse = (array) => {
+  array.sort((a,b) => a-b)
+   if(array[0] === array[1] && array[1] === array[2] && array[3] === array[4]){
+     return true
+   } else {
+    return false
+   }
+  }
```
Comment (elysemontano) [inline code-challenges.test.js:147]: I really like this approach!  One thing to consider is if the value in the pair is greater than or less than the three of a kind.  Currently, [5, 5, 3, 3, 3] will pass but [5, 5, 5, 4, 4] will fail.

File: interview-questions.md [hunk 3]
```diff
@@ -6,46 +6,46 @@ First, without external resources. Challenge yourself to answer from memory.
...
-Researched answer:
+Researched answer:A relational database has data stored in tables and it  knows relation between all data. 
 
 5. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: primary key help to differentiate between  two things that look similar.
 
-Researched answer:
+Researched answer:gives a unique ID for each set of data, it is important because it helps to avoid duplicates.
```
Comment (elysemontano) [inline interview-questions.md:37]: Awesome work on this section!

File: ruby-challenges.rb [hunk 4]
```diff
@@ -20,11 +28,41 @@
...
+  def brake num
+  @current_speed = @current_speed - num
+   if @current_speed < 0
+      @current_speed = 0
+   end
+  end
+end
+
+my_bike = Bike.new("Trek", 2, 0)
+p my_bike.getInfo
```
Comment (elysemontano) [inline ruby-challenges.rb:64]: 💯

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,42 +15,133 @@
...
 // b) Create the function that makes the test pass.
+
+
+
+
+
+
+
+
+const fullHouse = (array) => {
```
Comment (elysemontano) [inline code-challenges.test.js:140]: I don't see any pseudocode for this problem.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Requesting one change: Please provide pseudocode for problem 3 on code challenges.  Once this is added, I will approve this PR.
Other comments: Overall, you did a really great job on this assessment!  Your code is efficient and well written, interview answers were good and Ruby section was spot on.  To level up, try thinking about edge cases in your code and implementing tests and code that account for them.  Keep up the great work!

**Comment by Ahmedkhafaji** [comment]
I thought I did the pseudo code for number 3! did not see that sorry for the delay Elyse. thank you on the feed back, Noted!

File: code-challenges.test.js (hunk 5 above)
Comment (elysemontano) [reply code-challenges.test.js:140]: Great!  Thank you for adding that in!

**Review by elysemontano — APPROVED** [review]
Great work Ahmed!
